Add WooCommerce Payments Inbox note

thenbrent opened the pull request:
Fixes #4012

This PR adds a note to make merchants aware of WooCommerce Payments as a payment method. The note will appear on stores that:

* are in the US
* do not have the WooCommerce Payments plugin installed
* completed registration or the task list 7 days or more ago.

The note will only be added after April 14th - a week after the estimated WC 4.1 beta date, and a safe bet for the public beta date of WooCommerce Payments (the date the WooCommerce Payments plugin is published on WordPress.org).

**Please note**: this PR will have a merge conflict with SHA: 25f1124ffdb2160a6d, which is being submitted as part of #3978. I've committed the conflicting code separately in SHA: 66faccb to make that easier to address (that commit can be removed.)

### Screenshots

![Image upload on 2020-03-27 at 15-54-41](https://user-images.githubusercontent.com/235523/77750551-b7780b80-706f-11ea-92ed-1236d9bc2682.png)

### Detailed test instructions:

1. Checkout this branch
1. Trigger the `wc_admin_daily` hook scheduled in WP Cron 
1. Go to **Tools > Scheduled Actions**
1. Confirm the `wc-admin-woocommerce-payments_add_note` hook is scheduled for April 14
1. Trigger the `wc-admin-woocommerce-payments_add_note` hook
1. Go to a WooCommerce page
1. Click the Inbox icon in the Navbar
1. Observe the new **Try the new way to get paid** note
1. Click the **Install Now** button to test plugin installation

**Note:** if you want installation to work before the WC Pay plugin is published in April, you'll need to change the `WC_Admin_Notes_WooCommerce_Payments::PLUGIN_SLUG` and `WC_Admin_Notes_WooCommerce_Payments::PLUGIN_FILE` to link to a publicly available plugin that is already included in `Onboarding::get_allowed_plugins()`.

/cc @allendav @LevinMedia @claraclee

Commit:
Add WC Pay Note to Inbox

Commit:
Add WC Pay to allowed plugins

Commit:
Don't update WC Pay note content

Actioned notes are hidden from the Inbox it turns out, so there's
no need to update the content to reflect that the WC Pay plugin
has been installed.

Commit:
Add WC pay note only after April 14

Instead of only adding the Install Now button in April. Only
add the note so that the Install now button is always available
and generally, the note is more useful and actionable because
the plugin is actually available.

Commit:
Fix WC Pay note install action

timmyc commented on src/Notes/WC_Admin_Notes_WooCommerce_Payments.php:
```diff
@@ -0,0 +1,145 @@
+	 */
+	protected static function validate_plugin() {
+		include_once ABSPATH . '/wp-admin/includes/plugin.php';
+		return 0 === validate_plugin( self::PLUGIN_FILE );
```
While likely an edge case, if the plugin is installed in a differently named directory, this check might return false if wcpay is installed. Is there a constant, or a method name distinct to wcpay that we can use to sniff out if it is installed with instead?

thenbrent replied:
@timmyc thanks for the review. I've added SHA: 96e9e7d4 to check against the main `WC_Payments` class in addition to `validate_plugin()` (I also renamed the method given the double check and to better indicate it returns a boolean).

The method still checks against `validate_plugin( self::PLUGIN_FILE )` if the constant isn't defined, because we want to mark the notification as actioned even if the plugin has been installed and subsequently deactivated before `'wc_admin_daily'` is run again.

Commit:
Check if WC Pay is active or installed

Not just if the plugin files are available. This addresses both the
case where the plugin may be active an in a different folder name,
and where the plugin may have been installed but is not active,
likely because it's been deactivated.

becdetat commented on src/Notes/WC_Admin_Notes_WooCommerce_Payments.php:
```diff
@@ -0,0 +1,148 @@
+	const NOTE_NAME = 'wc-admin-woocommerce-payments';
+
+	/**
+	 * Name of the note for use in the database.
```
These comments are wrong

becdetat reviewed (approved):
This looks good to me, pending @timmyc being happy with the fix to the edge case and a minor note I made in the code

timmyc reviewed (approved):
LGTM
